src/transform/FooterReadMore.js

```diff
@@ -0,0 +1,349 @@
+    outerAnchorContainer.className = 'pagelib_footer_readmore_page'
+
+    const hasImage = wmfPage.thumbnail && wmfPage.thumbnail.source
+    if (hasImage) {
+      const image = document.createElement('div')
+      image.style.backgroundImage = `url(${wmfPage.thumbnail.source})`
+      image.classList.add('pagelib_footer_readmore_page_image')
+      outerAnchorContainer.appendChild(image)
+    }
+
+    const innerDivContainer = document.createElement('div')
+    innerDivContainer.classList.add('pagelib_footer_readmore_page_container')
+    outerAnchorContainer.appendChild(innerDivContainer)
+    outerAnchorContainer.href = `/wiki/${encodeURI(wmfPage.title)}`
+
+    if (wmfPage.title) {
+      const title = document.createElement('div')
+      title.id = index
+      title.className = 'pagelib_footer_readmore_page_title'
+      const displayTitle = wmfPage.title.replace(/_/g, ' ')
+      title.innerHTML = displayTitle
+      outerAnchorContainer.title = displayTitle
+      innerDivContainer.appendChild(title)
+    }
+
+    let description = null
+    if (wmfPage.terms) {
+      description = wmfPage.terms.description[0]
+    }
+    if ((description === null || description.length < 10) && wmfPage.extract) {
+      description = cleanExtract(wmfPage.extract)
+    }
+    if (description) {
+      const descriptionEl = document.createElement('div')
+      descriptionEl.id = index
+      descriptionEl.className = 'pagelib_footer_readmore_page_description'
+      descriptionEl.innerHTML = description
+      innerDivContainer.appendChild(descriptionEl)
+    }
+
+    const saveButton = document.createElement('div')
+    saveButton.id = `${_saveButtonIDPrefix}${encodeURI(wmfPage.title)}`
+    saveButton.innerText = _saveForLaterString
+    saveButton.title = _saveForLaterString
+    saveButton.className = 'pagelib_footer_readmore_page_save'
+    saveButton.addEventListener('click', event => {
+      event.stopPropagation()
+      event.preventDefault()
+      saveButtonClickHandler(wmfPage.title)
+    }, false)
+    innerDivContainer.appendChild(saveButton)
+
+    return document.createDocumentFragment().appendChild(outerAnchorContainer)
+  }
+}
+
+/**
+ * @type {ShownReadMoreHandler}
+ */
+const showReadMore = (pages, document, containerID, saveButtonClickHandler, titlesShownHandler) => {
+  const shownTitles = []
+  const container = document.getElementById(containerID)
+  pages.forEach((page, index) => {
+    const title = page.title.replace(/ /g, '_')
+    shownTitles.push(title)
+    const pageModel = new WMFPage(title, page.thumbnail, page.terms, page.extract)
+    const pageFragment = new WMFPageFragment(pageModel, index, document, saveButtonClickHandler)
+    container.appendChild(pageFragment)
+  })
+  titlesShownHandler(shownTitles)
+}
+
+/**
+ * Makes 'Read more' query parameters object for a title.
+ * @param {!string} title
+ * @param {!number} count
+ * @return {!Object}
+ */
+const queryParameters = (title, count) => ({
```

Comment:
This network request stuff might be worth breaking into a separate file, ReadMoreClient or something like that, so we can separate concerns between UI and networking easily.